### PR TITLE
[mle] add address outside of constructor

### PR DESCRIPTION
**Pull request by bukepo**
This PR adds link local address when MLE is enabled. Adding address in constructor gives no chance to notify address change with address callback.

### CODE REVIEW

File: src/core/net/netif.hpp [hunk 1]
```diff
@@ -447,6 +447,21 @@ class Netif : public InstanceLocator
...
+protected:
+    /**
+     * This method subscribes the network interface to the realm-local all MPL forwarders, link-local and
+     * realm-local all nodes address.
+     *
+     */
+    void SubscribeAllNodesMulticast(void);
+
+    /**
+     * This method unsubscribes the network interface to the realm-local all MPL forwarders, link-local and
```
Comment (jwhui) [inline src/core/net/netif.hpp]: nit: "interface to the" -> "interface from the"

File: src/core/thread/mle.cpp [hunk 2]
```diff
@@ -917,9 +925,13 @@ otError Mle::SetMeshLocalPrefix(const otMeshLocalPrefix &aMeshLocalPrefix)
     if (mRole == OT_DEVICE_ROLE_LEADER)
     {
         netif.RemoveUnicastAddress(mLeaderAloc);
-        memcpy(mLeaderAloc.GetAddress().mFields.m8, mMeshLocal64.GetAddress().mFields.m8, 8);
+        memcpy(mLeaderAloc.GetAddress().mFields.m8, aMeshLocalPrefix.m8, sizeof(aMeshLocalPrefix));
         netif.AddUnicastAddress(mLeaderAloc);
     }
+    else
+    {
+        memcpy(mLeaderAloc.GetAddress().mFields.m8, aMeshLocalPrefix.m8, sizeof(aMeshLocalPrefix));
```
Comment (librasungirl) [inline src/core/thread/mle.cpp]: `mLeaderAloc` may only be significant for FTD device. We may further enhance it in the further.
How about deferring to update its prefix when it is to be added in `Mle::AddLeaderAloc()`?

File: src/core/thread/mle.cpp [hunk 3]
```diff
@@ -868,7 +873,7 @@ otError Mle::SetMeshLocalPrefix(const otMeshLocalPrefix &aMeshLocalPrefix)
 {
     ThreadNetif &netif = GetNetif();
 
-    if (memcmp(mMeshLocal64.GetAddress().mFields.m8, aMeshLocalPrefix.m8, sizeof(aMeshLocalPrefix)) == 0)
+    if (memcmp(mLeaderAloc.GetAddress().mFields.m8, aMeshLocalPrefix.m8, sizeof(aMeshLocalPrefix)) == 0)
```
Comment (librasungirl) [inline src/core/thread/mle.cpp]: why here uses `mLeaderAloc`?

File: include/openthread/thread.h [hunk 4]
```diff
@@ -423,7 +423,7 @@ OTAPI const otMeshLocalPrefix *OTCALL otThreadGetMeshLocalPrefix(otInstance *aIn
  * @param[in]  aMeshLocalPrefix  A pointer to the Mesh Local Prefix.
  *
  * @retval OT_ERROR_NONE           Successfully set the Mesh Local Prefix.
- * @retval OT_ERROR_INVALID_STATE  Thread protocols are enabled.
+ * @retval OT_ERROR_INVALID_STATE  Thread protocols are not enabled.
```
Comment (librasungirl) [inline include/openthread/thread.h]: Keep it as is?
Once thread protocols are enabled, Mesh Local Prefix should only be updated through dataset, but not allowed to change via API.

**Pull request merged**